### ISSUE TITLE
TaskFlow: How to start a single task

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#10388]
Here's a question I can't seem to figure out on my own: For our own `Threads::new_task()` function, what I want to do is in essence what `std::async` does -- create a task, hand it off to the powers to be, walk away and return a handle so that someone can later check whether the task has finished.

I can't see how TaskFlow does that. Their executor takes a `tf::TaskFlow` objects, which of course I know how to create and fill with my task. But all of their examples then call `executor.run(taskflow).wait()`, i.e., they explicitly wait for the taskflow to finish.

Can I just call `executor.run(taskflow)` and then let the `taskflow` object go out of scope? Presumably yes, because `executor.run` returns a `std::future`. But are we sure that `executor.run` copies everything out of the taskflow object that is necessary?

**Comment by tjhei** [comment]
> Can I just call `executor.run(taskflow)` and then let the `taskflow` object go out of scope?

My understanding is that the taskflow object needs to be alive: 
> It is your responsibility to keep tasks and graph alive during their execution.
see https://cpp-taskflow.github.io/cpp-taskflow/chapter1.html#C1_LifetimeOfATask

**Comment by bangerth** [comment]
Ugh. That's awkward. I wouldn't know where to put the taskflow object. (I guess I could stick it into our `Task` object...) I'll ask on their forum.

**Comment by bangerth** [comment]
See https://github.com/cpp-taskflow/cpp-taskflow/issues/172

**Comment by tjhei** [comment]
> I guess I could stick it into our `Task` object...

that wouldn't be too bad, I would think. Do you want to play with this or do you want me to take a look?

**Comment by bangerth** [comment]
No worries, I know how to do it. But only after #10389.

**Comment by bangerth** [comment]
This could have been done using TaskFlow 2.7 and later, but we decided not to go with TaskFlow for now.
